sources/engine/Stride.Engine/Engine/Splines/Processors/SplineTransformProcessor.cs

```diff
@@ -0,0 +1,117 @@
+//// Copyright (c) Stride contributors (https://Stride.com)
+//// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Stride.Core.Mathematics;
+using Stride.Graphics;
+using Stride.Rendering;
+using Stride.Engine.Splines.Components;
+using Stride.Engine.Splines.HierarchyTransformOperations;
+
+namespace Stride.Engine.Splines.Processors
+{
+    /// <summary>
+    /// The processor for <see cref="SplineComponent"/>.
+    /// </summary>
+    public class SplineTransformProcessor : EntityProcessor<SplineComponent, SplineTransformProcessor.SplineTransformationInfo>
+    {
+
+        private SplineComponent SplineComponent;
+
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SplineTransformProcessor"/> class.
+        /// </summary>
+        public SplineTransformProcessor()
+            : base(typeof(TransformComponent))
+        {
+        }
+
+        protected override SplineTransformationInfo GenerateComponentData(Entity entity, SplineComponent component)
+        {
+            return new SplineTransformationInfo
+            {
+                TransformOperation = new SplineViewHierarchyTransformOperation(component),
+            };
+        }
+
+        protected override bool IsAssociatedDataValid(Entity entity, SplineComponent component, SplineTransformationInfo associatedData)
+        {
+            return component == associatedData.TransformOperation.SplineComponent;
+        }
+
+        protected override void OnEntityComponentAdding(Entity entity, SplineComponent component, SplineTransformationInfo data)
+        {
+            // Register model view hierarchy update
+            SplineComponent = component;
+
+            entity.Transform.PostOperations.Add(data.TransformOperation);
+        }
+
+        protected override void OnEntityComponentRemoved(Entity entity, SplineComponent component, SplineTransformationInfo data)
+        {
+            // Unregister model view hierarchy update
+            entity.Transform.PostOperations.Remove(data.TransformOperation);
+        }
+
+        public class SplineTransformationInfo
+        {
+            public SplineViewHierarchyTransformOperation TransformOperation;
+        }
+
+        public override void Draw(RenderContext context)
+        {
+            if (!SplineComponent.Spline.Dirty)
+            {
+                return;
+            }
```

Comment:
Hey, so the bug with only the last spline being rendered is because you only process the last one added to the `SplineTransformProcessor`, see lines attached.

At line 45 you keep a reference to the latest component added to the processor and at 63 you only consider that one for drawing.
A processor should process every component sent to it through the `OnEntityComponentX` pair of function. What I did to fix your issue is have add a `HashSet<SplineComponent> _components = new();` to the class, add and remove to and from that collection inside of the `OnEntityComponentX` pair of function and change the first couple of lines of draw to the following:
```cs
public override void Draw(RenderContext context)
{
    foreach (var SplineComponent in _components)
    {
        if (!SplineComponent.Spline.Dirty)
```
As for the issue with initial setup not drawing the spline, you should always force draw those that were just added to the processor.

Also you might want to be careful with how `Spline.Dirty` is used, if users read and write to it for whatever reason the transform processor won't draw/update as expected. You could fix that by adding an event to splines that the processor listens to, the processor could then push splines that fired those events to a collection for them to be redrawn next `Draw()`. It even has an added performance bonus since you won't need to iterate through all splines to check for dirty, just go through that collection of dirty splines and empty it after  you're finished updating/drawing them.

Reply:
> 
@Eideren  Thanks for your help and feedback.

It is so incredibly obvious now that you pointed it out. Using a hashset to keep track of all components that need updating is the way to go. I am also adding components to the hashset when they are dirty. Using an event for this is so much cleaner. I will do the same thing for the traverser component.